embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogOrchestrator.kt

```diff
@@ -0,0 +1,72 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import androidx.annotation.VisibleForTesting
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class LogOrchestrator(
+    private val logOrchestratorScheduledWorker: ScheduledWorker,
+    private val clock: Clock,
+    private val sink: LogSinkImpl
+) {
+    private var lastLogTime: Long = 0
+    private var inactivityFuture: ScheduledFuture<*>? = null
+    private var batchTimeFuture: ScheduledFuture<*>? = null
+
+    init {
+        sink.callOnLogsStored(::onLogsAdded)
+    }
+    private fun onLogsAdded() {
+        lastLogTime = clock.now()
+        if (sink.completedLogs().size >= MAX_LOGS_PER_BATCH) {
+            sendLogs()
+        } else {
+            scheduleInactivityCheck()
+            if (batchTimeFuture == null) {
+                scheduleBatchTimeCheck()
+            }
+        }
+    }
+
+    @VisibleForTesting
+    fun sendLogs() {
+        inactivityFuture?.cancel(false)
+        batchTimeFuture?.cancel(false)
+
+        val storedLogs = sink.flushLogs()
+
+        if (storedLogs.isNotEmpty()) {
+            // TBD: Send logs to DeliveryService
+        }
+    }
+
+    private fun scheduleInactivityCheck() {
+        inactivityFuture?.cancel(false)
+        inactivityFuture = logOrchestratorScheduledWorker.schedule<Unit>(
+            {
+                if (clock.now() - lastLogTime > MAX_INACTIVITY_TIME) {
```

Comment:
Is this verification needed? Why calling sendLogs() directly isn't possible given we are cancelling previous jobs and scheduling one for MAX_INACTIVITY_TIME from the time the log arrived.